contrib_bots/lib/gmail.py

```diff
@@ -0,0 +1,153 @@
+        email = create_message(body, subject, recipient)
+        result = send_message(service, email)
+
+        label_result = -1
+        if label:
+            create_label(service, label)
+            label_id = get_label_id(service, label)
+            label_result = add_label(service, label_id, result['id'])
+
+        if 'SENT' in result['labelIds']:
+            content = 'Email sent to {}'.format(recipient)
+            if label_result:  # Enters this block if the label couldn't be added
+                              # or if there was no label
+                content += ' without label.'
+            else:
+                content += '.'
+        else:
+            content = 'Could not send email to {}. Please try again.'.format(recipient)
+
+        client.send_message(dict(
+            type = 'private',
+            to = original_sender,
+            sender = self.bot_email,
+            content = content
+        ))
+
+def parse_input(user_input):
+    # Input is of the following format: (--label flag is optional)
+    # @gmail --recipient <recipient_email> --subject <subject> --label <label> --body <body>
+    if '--label' not in user_input:
+        loc = user_input.index('--body')
+        user_input = user_input[:loc] + '--label ' + user_input[loc:]
+
+    regex = "--recipient (.*?) --subject (.*?) --label (.*?) --body (.*?)\Z"
+    return re.findall(regex, user_input)[0]
+
+def create_message(body, subject, recipient):
+    message = MIMEText(body)
+    message['to'] = recipient
+    message['from'] = 'me'
+    message['subject'] = subject
+
+    raw = base64.urlsafe_b64encode(message.as_bytes())
+    return {'raw': raw.decode()}
+
+
+def send_message(service, email):
+    try:
+        result = service.users().messages().send(userId='me', body=email).execute()
+        return result
+    except errors.HttpError:
+        logging.exception('An error occured while sending the email.')
+        return -1
+
+
+def create_label(service, label):
+    # Creates the label if it doesn't already exist
+    label_object = {'name': label}
+    labels = service.users().labels().list(userId='me').execute()
+    labels = labels['labels']
+
+    if label not in [l['name'] for l in labels]:
+        service.users().labels().create(userId='me',
+                                        body=label_object).execute()
+
+def get_label_id(service, label_name):
+    labels = service.users().labels().list(userId='me').execute()
+
+    for l in labels['labels']:
+        if label_name == l['name']:
+            return l['id']
+
+def add_label(service, label_Id, msg_Id):
+    try:
+        label_object = {'addLabelIds': [label_Id]}
+        service.users().messages().modify(userId='me', id=msg_Id,
+                                          body=label_object).execute()
+        return 0
+    except errors.HttpError:
+        logging.exception('Encountered an error while adding the label to the message')
+        return -1
```

Comment:
I think it would be more readable to return `False` or `None` in case of a failure and `True` in case of a success.